crates/bevy_app/src/app_builder.rs

```diff
@@ -223,6 +303,22 @@ impl AppBuilder {
         self
     }
 
+    /// Inserts a main thread local resource to the app
+    ///
+    /// You usually want to use `insert_resource`, but there are some special cases when a resource must
+    /// be main-thread local.
+    ///
+    /// ## Example
+    /// ```
+    /// # use bevy_app::prelude::*;
+    /// #
+    /// struct State {
+    ///     counter: usize,
+    /// }
+    ///
+    /// App::build()
+    ///     .insert_thread_local_resource(State { counter: 0 });
+    /// ```
     pub fn insert_thread_local_resource<T>(&mut self, resource: T) -> &mut Self
```

Comment:
This isn't your fault, but the name of this function is wrong. We should verify and correct this class of errors.

Reply:
Seems like the name has now been changed to `insert_non_send_resource`, updated pull req as well